src/uucore/src/lib/mods/clap_localization.rs

```diff
@@ -120,167 +105,209 @@ pub fn handle_clap_error_with_exit_code(err: Error, util_name: &str, exit_code:
+
+        std::process::exit(exit_code);
+    } else {
+        // Generic fallback case
+        let error_word = translate!("common-error");
+        let colored_error_word = maybe_colorize(&error_word, Color::Red);
+        eprintln!("{colored_error_word}: unexpected argument");
+        std::process::exit(exit_code);
+    }
+}
 
-                // Get localized error word with fallback
-                let error_word = translate!("common-error");
+/// Handle InvalidValue and ValueValidation errors with localization
+fn handle_invalid_value_error(
+    err: Error,
+    util_name: &str,
+    maybe_colorize: impl Fn(&str, Color) -> String,
+) -> ! {
+    // Force localization initialization
+    crate::locale::setup_localization(util_name).ok();
+
+    // Extract value and option from error context using clap's context API
+    // This is much more robust than parsing the error string
+    let invalid_arg = err.get(ContextKind::InvalidArg);
+    let invalid_value = err.get(ContextKind::InvalidValue);
+
+    if let (Some(arg), Some(value)) = (invalid_arg, invalid_value) {
+        let option = arg.to_string();
+        let value = value.to_string();
+
+        // Check if this is actually a missing value (empty string)
+        if value.is_empty() {
+            // This is the case where no value was provided for an option that requires one
+            let error_word = translate!("common-error");
+            eprintln!(
+                "{}",
+                translate!("clap-error-value-required", "error_word" => error_word, "option" => option)
+            );
+            eprintln!();
+            eprintln!("{}", translate!("common-help-suggestion"));
+            std::process::exit(1);
+        } else {
+            // Get localized error word and prepare message components outside conditionals
+            let error_word = translate!("common-error");
+            let colored_error_word = maybe_colorize(&error_word, Color::Red);
+            let colored_value = maybe_colorize(&value, Color::Yellow);
+            let colored_option = maybe_colorize(&option, Color::Green);
+
+            let error_msg = translate!(
+                "clap-error-invalid-value",
+                "error_word" => colored_error_word,
+                "value" => colored_value,
+                "option" => colored_option
+            );
+
+            // For ValueValidation errors, include the validation error in the message
+            if matches!(err.kind(), ErrorKind::ValueValidation) {
+                if let Some(source) = err.source() {
+                    // Print error with validation detail on same line
+                    eprintln!("{error_msg}: {}", source);
+                } else {
+                    // Print localized error message
+                    eprintln!("{error_msg}");
+                }
+            } else {
+                // Print localized error message
+                eprintln!("{error_msg}");
+            }
```

Comment:
I think you can simplify it with something like:
```suggestion
            match err.source() {
                Some(source) if matches!(err.kind(), ErrorKind::ValueValidation) => {
                    eprintln!("{error_msg}: {source}");
                }
                _ => eprintln!("{error_msg}"),
            }
```